spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/catalog/SimpleFunctionRegistry.java

```diff
@@ -940,8 +942,15 @@ else if (ObjectUtils.isArray(output) && !(output instanceof byte[])) {
 				convertedOutput = this.convertMultipleOutputValuesIfNecessary(output, ObjectUtils.isEmpty(contentType) ? null : contentType);
 			}
 			else {
-				convertedOutput = messageConverter.toMessage(output,
-						new MessageHeaders(Collections.singletonMap(MessageHeaders.CONTENT_TYPE, contentType[0])));
+				if (headers == null) {
+					headers = new MessageHeaders(Collections.singletonMap(MessageHeaders.CONTENT_TYPE, contentType[0]));
+				}
+				else if (!ObjectUtils.isEmpty(contentType)) {
+					Map<String, Object> headersMap = (Map<String, Object>) ReflectionUtils
```

Comment:
Here i always over write the CONTENT_TYPE of the message. Because if i dont do so, some test out of BeanFactoryAwareFunctionRegistryTests, i dont understand will fail.
But in general i thin, the CONTENT_TYPE  of message should win to enable different encoder for the individual messages.